src/arch-clap.ts

```diff
@@ -0,0 +1,518 @@
+/* eslint-disable no-console, global-require, no-magic-numbers, max-statements */
+import * as Path from "path";
+import * as xsh from "xsh";
+import * as Fs from "fs";
+import * as filterScanDir from "filter-scan-dir";
+import * as xclap from "xclap";
+import * as _ from "lodash";
+
+const typeScriptDevDeps = {
+  // code coverage
+  "@istanbuljs/nyc-config-typescript": "^1.0.1",
+  "source-map-support": "^0.5.16",
+  // types for node.js
+  "@types/node": "^13.7.6",
+  // compilers
+  "ts-node": "^8.6.2",
+  typescript: "^3.8.3"
+};
+
+const eslintDevDeps = {
+  "babel-eslint": "^10.1.0",
+  eslint: "^6.8.0",
+  "eslint-config-walmart": "^2.2.1",
+  "eslint-plugin-filenames": "^1.1.0",
+  "eslint-plugin-jsdoc": "^21.0.0"
+};
+
+const eslintTSDevDeps = {
+  // eslint typescript deps
+  "@typescript-eslint/eslint-plugin": "^2.21.0",
+  "@typescript-eslint/parser": "^2.21.0"
+};
+
+/**
+ * User configurable options for @xarc/module-dev tasks
+ */
+export type XarcModuleDevOptions = {
+  /** force terminal colors in output - *default* `true` */
+  forceColor?: boolean;
+  /** Specify a XClap instance to use - *default* `require("xclap")` */
+  xclap?: object;
+  /** turn off/on linting tasks (using eslint) - *default* `true` */
+  enableLinting?: boolean;
+  /** Specify typescript config to override the default one */
+  tsConfig?: object;
+};
+
+/**
+ * setup PATH
+ */
+function setupPath(): void {
+  xsh.envPath.addToFront(Path.resolve("node_modules/.bin"));
+  xsh.envPath.addToFront(Path.join(__dirname, "../node_modules/.bin"));
+}
+
+/**
+ * Sort property keys of an object
+ *
+ * @param obj object
+ * @returns obj
+ */
+function sortObjKeys(obj: any): any {
+  return Object.keys(obj)
+    .sort()
+    .reduce((newObj: any, key: string) => {
+      newObj[key] = obj[key];
+      return newObj;
+    }, {});
+}
+
+/**
+ * read app's package.json
+ *
+ * @returns package JSON data
+ */
+function readAppPkgJson(): any {
+  return JSON.parse(Fs.readFileSync(Path.resolve("package.json")).toString());
+}
+
+/**
+ * write app's package.json with pkg
+ *
+ * @param pkg pkg data to write
+ */
+function writeAppPkgJson(pkg: any): void {
+  const data = JSON.stringify(pkg, null, 2);
+  Fs.writeFileSync(Path.resolve("package.json"), `${data}\n`);
+}
+
+class XarcModuleDev {
+  appPkg: any;
```

Comment:
`any` -> `{}`.

Reply:
I would also mark these fields as `readonly` if possible.

Reply:
tsc complains that `Property 'devDependencies' does not exist on type '{}'.ts(2339)`

the class sets this field to the app's package.json.  `readonly` probably won't work.